test/tests/functional/pbs_test_tpp.py

```diff
@@ -825,6 +829,149 @@ def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
                           job=True, interactive=True, resv=True,
                           resv_job=True)
 
+    def start_gdb(self, comm_pid):
+        """
+        This function starts gdb and attach it to pbs_comm
```

Comment:
this function is actually using gdb to count the number of comm threads.
I would suggest using "ps -eT | grep pbs_com[m] | wc -l" . subtract 1 from the result.